components/formats-api/src/loci/formats/FormatTools.java

```diff
@@ -405,6 +427,42 @@ public static int getIndex(String order, int zSize, int cSize, int tSize,
   }
 
   /**
+   * Gets the rasterized index corresponding to the given Z, C, T,
+   * ModuloZ, ModuloC, ModuloT coordinates.
+   *
+   * @param order Dimension order.
+   * @param zSize Total number of focal planes.
+   * @param cSize Total number of channels.
+   * @param tSize Total number of time points.
+   * @param moduloZSize Total number of ModuloZ planes.
+   * @param moduloCSize Total number of ModuloC planes.
+   * @param moduloTSize Total number of ModuloT planes.
+   * @param num Total number of image planes (zSize * cSize * tSize),
+   *   specified as a consistency check.
+   * @param z Z coordinate of ZCTmZmCmT coordinate sextuple to convert to 1D index.
+   * @param c C coordinate of ZCTmZmCmT coordinate sextuple to convert to 1D index.
+   * @param t T coordinate of ZCTmZmCmT coordinate sextuple to convert to 1D index.
+   * @param moduloZ ModuloZ coordinate of ZCTmZmCmT coordinate sextuple to convert to 1D index.
+   * @param moduloC ModuloC coordinate of ZCTmZmCmT coordinate sextuple to convert to 1D index.
+   * @param moduloT ModuloT coordinate of ZCTmZmCmT coordinate sextuple to convert to 1D index.
+   */
+  public static int getIndex(String order,
+    int zSize, int cSize, int tSize,
+    int moduloZSize, int moduloCSize, int moduloTSize,
+    int num,
+    int z, int c, int t,
+    int moduloZ, int moduloC, int moduloT) {
+    return getIndex(order,
+                    zSize,
+                    cSize,
+                    tSize,
+                    num,
+                    (z * moduloZSize) + moduloZ,
+                    (c * moduloCSize) + moduloC,
+                    (t * moduloTSize) + moduloT);
```

Comment:
OK, but then you will need to fix lines 336-338, which are definitely not passing through the effective sizes.  Updating the Javadocs as well would certainly be helpful, as all of the other related methods are working with the absolute size Z/C/T.